File: WarGame/src/app/data/PlayerSlot.ts
export class PlayerSlot {
    public seat: number;
    public playerId: string;
    public connectionId: string;
    public name: string;

    constructor(seat: number, playerId: string, connectionId: string, name: string) {
        this.seat = seat;
        this.playerId = playerId;
        this.connectionId = connectionId;
        this.name = name;
    }
}

File: WarGameServer/WarGameServer/Data/PlayerSlot.cs
public class PlayerSlot
{
    public PlayerSlot(int seat, string playerId, string connectionId, string name)
    {
        Seat = seat;
        PlayerId = playerId;
        ConnectionId = connectionId;
        Name = name;
    }

    public int Seat { get; }

    public string PlayerId { get; }

    public string? ConnectionId { get; set; }

    public string Name { get; }

    public PlayerSlot WithConnection(string connectionId) {
        return new PlayerSlot(Seat, PlayerId, connectionId, Name);
    } 
}